sympy/solvers/diophantine/diophantine.py

```diff
@@ -3742,14 +3762,17 @@ def sum_of_four_squares(n):
     See Also
     ========
 
-    sum_of_squares()
+    power_representation :
+        ``sum_of_four_squares(n)`` is one of the solutions output by ``power_representation(n, 2, 4, zeros=True)``
+
     """
+    n = as_int(n)
+    if n < 0:
+        raise ValueError("n should be a non-negative integer")
     if n == 0:
         return (0, 0, 0, 0)
-
-    v = multiplicity(4, n)
-    n //= 4**v
-
+    n, v = remove(n, 4)
```

Comment:
```suggestion
    # remove factors of 4 since a solution in terms of 3 squares is
    # going to be returned; this is also done in sum_of_three_squares,
    # but it needs to be done here to select d
    n, v = remove(n, 4)
```